### PR TITLE
[ApiBundle] Unify of section provider

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | #16097
| License         | MIT

### CODE REVIEW

**Review by mpysiak — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Doctrine/ORM/QueryExtension/Admin/Promotion/PromotionCoupon/PostResultExtension.php [hunk 1]
```diff
@@ -18,11 +18,18 @@
...
 use Sylius\Component\Core\Model\PromotionCouponInterface;
 
 /** @implements QueryResultItemExtensionInterface<PromotionCouponInterface> */
 final readonly class PostResultExtension implements QueryResultItemExtensionInterface
 {
+    public function __construct(private SectionProviderInterface $sectionProvider)
+    {
+    }
+
+
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Doctrine/ORM/QueryExtension/Admin/Promotion/PromotionCoupon/PostResultExtension.php:32]:
```suggestion

```

File: src/Sylius/Bundle/ApiBundle/Doctrine/ORM/QueryExtension/Admin/Promotion/PromotionCoupon/PostResultExtension.php [hunk 2]
```diff
@@ -40,7 +47,9 @@ public function applyToItem(
     /** @param array<array-key, mixed> $context */
     public function supportsResult(string $resourceClass, ?Operation $operation = null, array $context = []): bool
     {
-        return $operation instanceof Post && is_a($resourceClass, PromotionCouponInterface::class, true);
+        return $operation instanceof Post
+            && $this->sectionProvider->getSection() instanceof AdminApiSection
+            && is_a($resourceClass, PromotionCouponInterface::class, true);
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Doctrine/ORM/QueryExtension/Admin/Promotion/PromotionCoupon/PostResultExtension.php:52]:
```suggestion
        return 
            $operation instanceof Post &&
            $this->sectionProvider->getSection() instanceof AdminApiSection &&
            is_a($resourceClass, PromotionCouponInterface::class, true)
        ;
```

File: src/Sylius/Bundle/ApiBundle/Serializer/Normalizer/ProductVariantNormalizer.php [hunk 3]
```diff
@@ -97,16 +102,14 @@ public function supportsNormalization(mixed $data, ?string $format = null, array
             return false;
         }
 
-        return $data instanceof ProductVariantInterface && $this->isNotAdminApiSection();
+        return $data instanceof ProductVariantInterface &&
+            $this->uriBasedSectionContext->getSection() instanceof ShopApiSection &&
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Serializer/Normalizer/ProductVariantNormalizer.php:106]:
```suggestion
        return 
            $data instanceof ProductVariantInterface &&
            $this->uriBasedSectionContext->getSection() instanceof ShopApiSection &&
```

File: src/Sylius/Bundle/ApiBundle/spec/Serializer/Normalizer/GeneratedPromotionCouponsNormalizerSpec.php [hunk 4]
```diff
@@ -136,13 +147,35 @@ public function it_throws_an_exception_if_it_is_not_admin_section(
...
+            ->shouldNotBeCalled()
+        ;
+
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('normalize', [new ArrayCollection([$promotionCoupon]), null, ['groups' => ['sylius:promotion_coupon:index']]])
+        ;
+    }
+
+    public function it_throws_an_exception_if_serializer_group_is_not_supported(
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/spec/Serializer/Normalizer/GeneratedPromotionCouponsNormalizerSpec.php:161]:
```suggestion
    public function it_throws_an_exception_if_serialization_group_is_not_supported(
```
🤔

**Review by GSadee — APPROVED** [review]

**Pull request merged**